backend/db/utils.py

```diff
@@ -218,9 +220,31 @@ def refresh_derived_tables(
             break
 
 
+# @joeflack4: finally put in connection pooling. (it didn't help the problem I was having)
+# got the code from our friend: https://chat.openai.com/share/0472ba44-0e58-44ad-bbcb-5eabffae3d39
+def set_search_path(dbapi_connection, connection_record):
+    """Set the search_path when connecting."""
+    if SCHEMA:
+        cursor = dbapi_connection.cursor()
+        cursor.execute(f"SET SESSION search_path='{SCHEMA}'")
+        cursor.close()
+
+def get_db_connection(isolation_level='AUTOCOMMIT', schema: str = SCHEMA, local=False) -> Connection:
+    """Get database connection object"""
+    if schema == None:
```

Comment:
## Bug fixes: To new implementation of `get_db_connection()`

### Connecting to public schema
Siggie recently added a `ValueError` for `if not schema` in `get_db_connection()`. However, we have like 15 instances where we are connecting via `schema=''`. This is how we connect to `public`, I thought. I checked w/ ChatGPT and they say that `''` and `'public'` are not the same. However I'm not sure if I trust their response. If it ain't broke, don't fix it; so I want to keep allowing this.

@Sigfried Is there something that broke because schema was explicitly set to `None` or `''`? If so, then I have to maybe change all these instances to `'public'` and run the test suite again (which should hopefully show that nothing has broke).

## Schema not getting passed around correctly
`set_search_path()` was using `SCHEMA`, and didn't have a `schema` arg either.

## Refactor back to old method if we need
@Sigfried said that this new method didn't help them, so we can refactor back if needed.